src/util/mpir_hw_topo.c

```diff
@@ -0,0 +1,564 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#include "mpiimpl.h"
+#include "mpir_hw_topo.h"
+
+#ifdef HAVE_HWLOC
+#include "hwloc.h"
+#endif
+
+#ifdef HAVE_NETLOC
+#include "mpir_netloc.h"
+#endif
+
+/*
+ * Hardwre topology
+ */
+static struct {
+#ifdef HAVE_HWLOC
+    hwloc_topology_t hwloc_topology;
+    hwloc_cpuset_t bindset;
+#endif
+    int bindset_is_valid;
+#ifdef HAVE_NETLOC
+    netloc_topology_t netloc_topology;
+    MPIR_Netloc_network_attributes network_attr;
+#endif
+} hw_topo = {
+0};
+
+enum {
+    MPIR_OBJ_TO_HW,
+    HW_OBJ_TO_MPIR
+};
```

Comment:
Weird naming.  `MPIR_` is a prefix.  What is this enum for?

Reply:
It should not be there anymore. I thought I had removed it.